orchagent/tunneldecaporch.cpp

```diff
@@ -176,23 +176,41 @@ void TunnelDecapOrch::doTask(Consumer& consumer)
                 }
                 else if (fvField(i) == decap_dscp_to_tc_field_name)
                 {
-                    dscp_to_dc_map_id = gQosOrch->resolveTunnelQosMap(table_name, key, decap_dscp_to_tc_field_name, t); 
-                    if (exists && dscp_to_dc_map_id != SAI_NULL_OBJECT_ID)
+                    dscp_to_tc_map_id = gQosOrch->resolveTunnelQosMap(table_name, key, decap_dscp_to_tc_field_name, t);
+                    if (dscp_to_tc_map_id == SAI_NULL_OBJECT_ID)
                     {
-                        setTunnelAttribute(fvField(i), dscp_to_dc_map_id, tunnel_id);
+                        SWSS_LOG_NOTICE("QoS map %s is not ready yet", decap_dscp_to_tc_field_name.c_str());
+                        task_status = task_process_status::task_need_retry;
```

Comment:
I would suggest, please retain the previous logic and just return 'false' from here if its NULL. Since its only two cases that needs a retry.

Reply:
How can we handle the case when QoS table is specified for a tunnel, while the QoS table is not processed in `qosorch`?
I think it's possible that the update event of  `APP_TUNNEL_DECAP_TABLE_NAME` comes before QoS table, like `DSCP_TO_TC_MAP`.

Reply:
agree, my point is, retry is required only in this new qos "if-else" case. we can retain the logic for all other above cases.. You just need to return false from here.

Reply:
Fixed. Now the retry logic only exists for the qos section. Thanks